hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/log/LogThrottlingHelper.java

```diff
@@ -259,6 +259,10 @@ public LogAction record(String recorderName, long currentTimeMs,
       currentLogs.put(recorderName, currentLog);
     }
     currentLog.recordValues(values);
+    if (currentTimeMs < lastLogTimestampMs) {
+      // Reset lastLogTimestampMs: this should only happen in tests
+      lastLogTimestampMs = Long.MIN_VALUE;
+    }
```

Comment:
This is a bit awkward. How about instead we add a new method like `reset()` that clears all of the state?

Reply:
I thought about this solution, but the problem is that we have to add `reset()` methods in `LogThrottlingHelper` as well as `FSEditLogLoader`. Will this be OK?

Reply:
Why does `FSEditLogLoader` need a reset method? Just so we can trigger the reset from within `TestFSEditLogLoader`? If so then can we just made `FSEditLogLoader.LOAD_EDITS_LOG_HELPER` package-private so that we can access it from within `TestFSEditLogLoader` and then call `reset()` directly on the `LogThrottlingHelper`?

Reply:
Nice solution. I will fix this.